src/leader.py

```diff
@@ -59,10 +59,12 @@ def set_speed(self, position):
         :param position: The value of the button input.
         :type position: int
         """
-        if position == -1:
+        try:
+            position = int(position)
```

Comment:
doesn't this round the value of position to the nearest integer? Wouldn't that mess up the change in speed and only allow for motion in full speed, half speed and stopped?

Reply:
Yeah that's my mistake. I was just looking at the docstring. Fixed.